facebook/docusaurus
# feat(v2): add support specify new languages for Prism

## Motivation

Resolve #1879 and close #2356 as our users have an interest in this feature, I think we need to add this.

Related issues on Canny:
- https://docusaurus.canny.io/admin/board/feature-requests/p/support-rust-syntax-highlighting
- https://docusaurus.canny.io/admin/board/feature-requests/p/expand-default-syntax-highlighting-language-support

Cons of the current implementation:
- ~~Pollution of global space, I do not know how to avoid this, given the use of Prism React Renderer :(~~

### Have you read the [Contributing Guidelines on pull requests](https://github.com/facebook/docusaurus/blob/master/CONTRIBUTING.md#pull-requests)?

Yes

## Test Plan

1. Add `themeConfig.prism.additionalLanguages` array with any languages missing in Prism React Renderer (rust eg)
2. Add corresponding  code block, eg for rust:

```
```rust
fn main() {
    // Statements here are executed when the compiled binary is called

    // Print text to the console
    println!("Hello World!");
}
```

3. Make sure that it syntax highlighting for the added code block works correctly.

## Related PRs

(If this PR adds or changes functionality, please take some time to update the docs at https://github.com/facebook/docusaurus, and link to your PR here.)
